src/tree.ts

```diff
@@ -460,14 +512,18 @@ export class ProjectOutlineProvider implements vscode.TreeDataProvider<BaseNode>
 
   getChildren(node?: BaseNode): BaseNode[] {
     try {
-      if (!node) {
-        // Request for root node
-        return this._children;
-      } else {
+      if (node) {
         return node.getChildren();
       }
+      // Request for root nodes
+      if (this._folders.size <= 1) {
+        for (const folder of this._folders.values()) {
```

Comment:
I'm a little confused about the scenario that this covers.  I'll drop by and ask about it.